salt/fileclient.py

```diff
@@ -956,31 +956,31 @@ def get_file(self,
                 load['loc'] = 0
             else:
                 load['loc'] = fn_.tell()
-            data = self.channel.send(load)
+            data = self.channel.send(load, raw=True)
             try:
-                if not data['data']:
-                    if not fn_ and data['dest']:
+                if not data[b'data']:
+                    if not fn_ and data[b'dest']:
                         # This is a 0 byte file on the master
                         with self._cache_loc(
-                                data['dest'],
+                                data[b'dest'],
                                 saltenv,
                                 cachedir=cachedir) as cache_dest:
                             dest = cache_dest
                             with salt.utils.fopen(cache_dest, 'wb+') as ofile:
-                                ofile.write(data['data'])
-                    if 'hsum' in data and d_tries < 3:
+                                ofile.write(data[b'data'])
+                    if b'hsum' in data and d_tries < 3:
                         # Master has prompted a file verification, if the
                         # verification fails, re-download the file. Try 3 times
                         d_tries += 1
-                        hsum = salt.utils.get_hash(dest, data.get('hash_type', 'md5'))
-                        if hsum != data['hsum']:
+                        hsum = salt.utils.get_hash(dest, data.get(b'hash_type', 'md5'))
```

Comment:
Since you are using 'raw' mode, if 'hash_type' is in data, then its value will be stored as bytes instead of str. I believe that `salt.utils.get_hash` currently expects that arg to be a str (since it uses `getattr` and the docs say the name must be str). So perhaps you will need something like:

```
hsum = salt.utils.get_hash(dest, salt.utils.to_str(data.get(b'hash_type', b'md5')))
```